src/main/java/org/squiddev/plethora/integration/astralsorcery/IntegrationAstralSorcery.java

```diff
@@ -0,0 +1,777 @@
+
+			return out;
+		}
+
+		@Nullable
+		@Override
+		public TileCollectorCrystal getExample() {
+			return null;
+		}
+	};
+
+	public static final IMetaProvider<ItemStack> META_ITEM_COLLECTOR_CRYSTAL = new ItemStackContextMetaProvider<ItemCollectorCrystal>(
+		ItemCollectorCrystal.class,
+		"FIXME Set the description"
+	) {
+		@Nonnull
+		@Override
+		public Map<String, Object> getMeta(@Nonnull IPartialContext<ItemStack> context, @Nonnull ItemCollectorCrystal item) {
+			Map<String, Object> out = new HashMap<>(3);
+			ItemStack stack = context.getTarget();
+
+			//CrystalProperties of the item form are already exposed
+			IWeakConstellation mainConstellation = ItemCollectorCrystal.getConstellation(stack);
+			out.put("constellation", mainConstellation != null ? context.makePartialChild(mainConstellation).getMeta() : null);
+
+			IMinorConstellation traitConstellation = ItemCollectorCrystal.getTrait(stack);
+			out.put("traitConstellation", traitConstellation != null ? context.makePartialChild(traitConstellation).getMeta() : null);
+
+			out.put("crystalType", ItemCollectorCrystal.getType(stack).name());
+
+			return out;
+		}
+
+		@Nullable
+		@Override
+		public ItemStack getExample() {
+			return null;
+		}
+	};
+
+	public static final IMetaProvider<TileCrystalLens> META_TILE_CRYSTAL_LENS= new BaseMetaProvider<TileCrystalLens>() {
+
+		@Nonnull
+		@Override
+		public Map<String, Object> getMeta(@Nonnull IPartialContext<TileCrystalLens> context) {
+			Map<String, Object> out = new HashMap<>(3);
+			TileCrystalLens target = context.getTarget();
+
+			out.putAll(context.makePartialChild(target.getCrystalProperties()).getMeta());
+
+			ItemColoredLens.ColorType lens = target.getLensColor();
+			String colorName = lens != null ? lens.getUnlocalizedName() : null;
+
+			out.put("lensColor", colorName);
+			out.put("lensColour", colorName);
+
+			return out;
+		}
+
+		@Nullable
+		@Override
+		public TileCrystalLens getExample() {
+			return null;
+		}
+	};
+
+	public static final IMetaProvider<ItemStack> META_SKY_RESONATOR = new ItemStackContextMetaProvider<ItemSkyResonator>(
+		ItemSkyResonator.class,
+		"Provides the available modes for this Resonator"
+	) {
+		@Nonnull
+		@Override
+		public Map<String, Object> getMeta(@Nonnull IPartialContext<ItemStack> context, @Nonnull ItemSkyResonator item) {
+			//REFINE Do we want `getUnlocalizedName` or `getUnlocalizedUpgradeName`?
+			// Further, should we localize the String?
+
+			/*
+			ItemStack stack = context.getTarget();
+			List<ItemSkyResonator.ResonatorUpgrade> modes = ItemSkyResonator.getUpgrades(stack);
+			LuaList<String> modeNames = new LuaList<>(modes.size());
+
+			for (ItemSkyResonator.ResonatorUpgrade mode : modes) {
+				modeNames.add(mode.getUnlocalizedName());
+			}
+
+			return Collections.singletonMap("modes", modeNames.asMap()); */
+
+			return Collections.singletonMap("modes",
+				ItemSkyResonator.getUpgrades(context.getTarget()).stream()
```

Comment:
Might be able to do `LuaList.of(..., ItemSkyResonator.ResonatorUpgrade::getUnlocalizedName)` instead